### PR TITLE
Bump version to 0.17

### PR DESCRIPTION
**Pull request by qinsoon**

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,4 +1,57 @@
...
+    updating weak references. `Scanning::process_weak_refs()` allows a boolean return value to indicate if the method should be called again
+    by MMTk after finishing transitive closures for the weak references, which can be used to implement ephemeral weak references.
+  * Add `Collection::post_forwarding()` which is called by MMTk after all weak reference related work is done. A binding can use this call
+    for any post processing for weak references, such as enqueue cleared weak references to a queue (Java).
+  * These replace old methods for the same purpose, like `Collection::process_weak_refs()`, `Collection::vm_release()`, and `memory_manager::on_closure_end`.
+
+
+Misc
+---
+* Improve the boot time significantly for MMTk:
```
Comment (k-sareen) [inline CHANGELOG.md:41]: Should mention move to Rust v1.66.1.

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,4 +1,58 @@
...
+* Add a 'sentinel' mechanism to execute a specified 'sentinel' work packet when a bucket is drained, and prevent the GC from entering the next stage depending
+  on the sentinel packet. This makes it possible to expand the transitive closure multiple times. It replaces `GCWorkScheduler::closure_end()`.
+
+
+API
+---
+* Add a new set of APIs to support binding-specific weak reference processing:
+  * Add `Scanning::process_weak_refs()` and `Scanning::forward_weak_refs()`. They both supply an `ObjectTracerContext` argument for retaining and
+    updating weak references. `Scanning::process_weak_refs()` allows a boolean return value to indicate if the method should be called again
+    by MMTk after finishing transitive closures for the weak references, which can be used to implement ephemeral weak references.
```
Comment (wks) [inline CHANGELOG.md:33]: The return value here is intended to support "ephemeron".  It is a well-known term, and it is used in V8.  I can't find any use of the term "ephemeral weak reference".

```suggestion
    by MMTk after finishing transitive closures for the weak references, which can be used to implement ephemerons.
```

Comment (qinsoon) [reply CHANGELOG.md]: Done. Thanks.

**Review by wks — APPROVED** [review]
LGTM

**Pull request merged**